src/layman/map/util.py

```diff
@@ -166,31 +169,63 @@ def get_complete_map_info(workspace=None, mapname=None, cached=False):
     return complete_info
 
 
+def get_schema_url_patter():
+    if _SCHEMA_URL.get() is None:
+        _SCHEMA_URL.set(re.compile(SCHEMA_URL_PATTER))
+    return _SCHEMA_URL.get()
+
+
+def get_composition_schema(url):
+    match = get_schema_url_patter().match(url)
+    if not match:
+        raise LaymanError(2, {
+            'parameter': 'file',
+            'reason': 'Invalid schema url',
+            'regular_expression': SCHEMA_URL_PATTER,
+        })
+    schema_file_name = os.path.join(*url.split('/')[-1:])
+    schema_path = os.path.join(*url.split('/')[-3:-1])
+    local_path = os.path.join(_SCHEMA_CACHE_PATH, schema_path)
+    local_full_path = os.path.join(local_path, schema_file_name)
+    if os.path.exists(local_full_path):
+        with open(local_full_path) as schema_file:
+            schema_json = json.load(schema_file)
+    else:
+        res = requests.get(url,
+                           timeout=settings.DEFAULT_CONNECTION_TIMEOUT)
+        res.raise_for_status()
+        schema_txt = res.text
+        schema_json = json.loads(schema_txt)
+
+        os.makedirs(local_path, exist_ok=True)
+        out_file = open(local_full_path, "w")
+        json.dump(schema_json, out_file, indent=4)
+        out_file.close()
+    return schema_json
+
+
 def check_file(file):
     try:
         file_json = json.load(file)
-        schema_path = os.path.join(
-            os.path.dirname(os.path.realpath(__file__)),
-            'schema.draft-07.json'
-        )
-        with open(schema_path) as schema_file:
-            schema_json = json.load(schema_file)
-            validator = Draft7Validator(schema_json)
-            if not validator.is_valid(file_json):
-                errors = [
-                    {
-                        'message': e.message,
-                        'absolute_path': list(e.absolute_path),
-                    }
-                    for e in validator.iter_errors(file_json)
-                ]
-                raise LaymanError(2, {
-                    'parameter': 'file',
-                    'reason': 'JSON not valid against schema layman/map/schema.draft-07.json',
-                    'validation-errors': errors,
-                })
-            validate(instance=file_json, schema=schema_json)
-            return file_json
+        schema_url = file_json['describedBy']
```

Comment:
Ensure that if file is map composition v 1.0.0 or other JSON, that Layman raises LaymanError with appropriate message